build_tag_withcallstack.go

```diff
@@ -0,0 +1,27 @@
+// +build withcallstack
+
+//   Copyright © 2015-2017 Ivan Kostko (github.com/ivan-kostko; github.com/gopot)
+
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+
+//       http://www.apache.org/licenses/LICENSE-2.0
+
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package errors
+
+import (
+	"github.com/gopot/errors/detalizers"
+)
+
+// Resets Default ErrorFactory whith NewCallStackDetalizer
+func init() {
+
+	defaultErrorFactory = NewErrorFactory(NewDefaultKVStorage, detalizers.NewCallStackDetalizer(2, 1024))
```

Comment:
I read a nice article about package-level variables recently. Do you want a link ;)

Reply:
Yes, please. I bet in the article it was told to avoid package variables storing **state** , which is not a case here.

Reply:
[here](https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables)
You are right. But still you are changing `defaultErrorFactory` on init, which makes the code harder to understand and maintain.

Reply:
Actually it is done under build tag, which makes it clear when and how it would be applied, isnt it?